pylint/testutils.py

```diff
@@ -622,6 +625,11 @@ def _split_lines(cls, expected_messages, lines):
 
     def _check_output_text(self, expected_messages, expected_lines, received_lines):
         expected_lines = self._split_lines(expected_messages, expected_lines)[0]
-        assert (
-            expected_lines == received_lines
-        ), "Expected test lines did not match for test: {}".format(self._test_file.base)
+        for exp, rec in itertools.zip_longest(expected_lines, received_lines):
+            assert exp == rec, (
+                "Expected test lines did not match for test: {}."
+                "{_linesep:s}Expected : {}"
+                "{_linesep:s}Received : {}".format(
+                    self._test_file.base, exp, rec, _linesep=os.linesep
```

Comment:
Is `os.linesep` necessary ? What happen if you just put `\n` ?

Reply:
I'm afraid it would not work on non Unix-like system (Windows for example).

Reply:
So I did some research because I never had any problem with this and I found this : https://stackoverflow.com/questions/38074811/what-is-os-linesep-for It also looks like it's defined in pep278. I wonder if the print function is in "text mode", I would say yes, because I don't think it's working with bytes?  The documentation does not talk about print so it's probably warranted for a print https://docs.python.org/3/library/os.html#os.linesep.

This is just for your information btw, the code is working as it is let's keep it that way :)